rma/models/rma.py

```diff
@@ -440,27 +444,30 @@ def _onchange_user_id(self):
 
     @api.onchange("partner_id")
     def _onchange_partner_id(self):
+        self.picking_id = partner_invoice_id = False
         if self.partner_id:
             address = self.partner_id.address_get(['invoice'])
-            self.partner_invoice_id = address.get('invoice', False)
-        else:
-            self.partner_invoice_id = False
+            partner_invoice_id = address.get('invoice', False)
+        self.partner_invoice_id = partner_invoice_id
+        self.picking_id = False
 
     @api.onchange("picking_id")
     def _onchange_picking_id(self):
+        location = False
         if self.picking_id:
-            if not self.location_id:
-                warehouse = self.picking_id.picking_type_id.warehouse_id
-                self.location_id = warehouse.rma_loc_id.id
-        else:
-            self.move_id = self.product_id = self.location_id = False
+            warehouse = self.picking_id.picking_type_id.warehouse_id
+            location = warehouse.rma_loc_id.id
+        self.location_id = location
+        self.move_id = self.product_id = False
```

Comment:
I don't like too much multiple assignations in same line as they are less readable.